internal/internal_workflow_client.go

```diff
@@ -326,24 +326,7 @@ func (wc *workflowClient) SignalWorkflow(ctx context.Context, workflowID string,
 	if err != nil {
```

Comment:
nit: the encoding could be moved to common signal function

Reply:
I would prefer to keep it outside, at least with this PR.
It is consistent with other functions used there that way. I could refactor all of them with another PR if you like.